scripts/demo_common.sh

```diff
@@ -105,18 +101,33 @@ run_bad () {
 ## Assume an in-tree build.
 programs_dir="$root_dir/programs"
 
+has_query_compile_time_config () {
+  if ! [ -f "$1/programs/test/query_compile_time_config" ]; then
+    return 1;
+  fi
+  return 0;
+}
+
+if has_query_compile_time_config "$root_dir"; then
+  query_compile_time_config_dir="$programs_dir/test"
+elif is_valid_root "$root_dir/.." && has_query_compile_time_config "$root_dir/.."; then
+  query_compile_time_config_dir="$root_dir/../programs/test"
+else
+  query_compile_time_config_dir=""
+fi
+
 ## config_has SYMBOL...
 ## Succeeds if the library configuration has all SYMBOLs set.
 ##
 ## Note: "query_compile_time_config" is only available when in Mbed TLS project,
 ## so "config_has" is not available in TF-PSA-Crypto one. If this function is
 ## called from the latter we fail immediately.
```

Comment:
If `query_compile_time_config` is needed but absent (for example, if the executable was built out of tree), the demo script will silently be skipped. Is this really what we want?